src/BlockTypes/ProductGalleryThumbnails.php

```diff
@@ -37,7 +37,48 @@ protected function get_block_type_style() {
 	 * @return string[]
 	 */
 	protected function get_block_type_uses_context() {
-		return [ 'productGalleryClientId', 'postId', 'thumbnailsNumberOfThumbnails', 'thumbnailsPosition' ];
+		return [ 'productGalleryClientId', 'postId', 'thumbnailsNumberOfThumbnails', 'thumbnailsPosition', 'mode' ];
+	}
+
+	/**
+	 * Generate the View All markup.
+	 *
+	 * @param int $remaining_thumbnails_count The number of thumbnails that are not displayed.
+	 *
+	 * @return string
+	 */
+	protected function generate_view_all_html( $remaining_thumbnails_count ) {
+		$view_all_html = '<div class="wc-block-product-gallery-thumbnails__thumbnail__overlay wc-block-product-gallery-dialog-on-click" data-wc-on--click="actions.woocommerce.handleClick">
+			<span class="wc-block-product-gallery-thumbnails__thumbnail__remaining-thumbnails-count wc-block-product-gallery-dialog-on-click">+%1$s</span>
+			<span class="wc-block-product-gallery-thumbnails__thumbnail__view-all wc-block-product-gallery-dialog-on-click">%2$s</span>
+			</div>';
+
+		return sprintf(
+			$view_all_html,
+			esc_html( $remaining_thumbnails_count ),
+			__( 'View all', 'woo-gutenberg-products-block' )
```

Comment:
This needs escaping. `esc_html__()`

Reply:
Are you referring to `__( 'View all', 'woo-gutenberg-products-block' )`?

Could you elaborate further? I believe the escaping is sufficient here.

Reply:
Unless I am missing something, I am not seeing any escaping for that translator text.

Reply:
Right, I believe translated strings are generally considered safe in terms of output (unless when getting echo'd directly in an HTML attribute or a JavaScript context). In this specific case, it's used within the HTML context, so it should be safe.

Reply:
On the contrary, you should never trust the translator. Please see this reference https://developer.wordpress.org/reference/functions/esc_html__/

Reply:
Searching through our repo yields a lot of results of translated unescaped plain text strings. If these are considered unsafe, we should fix them.

I did some searching and found this very interesting thread on escaping every translated string: poqVs-c0T-p2

It is quite old though, so I wonder if anything has changed since then.

Reply:
Thanks @roykho, I really appreciate you taking the time to respond to this. I had a follow-up Slack discussion with @nerrad regarding escaping translated strings: p1697572533886259-slack-C02FL3X7KR6 - I'm happy to make the update :)